### PR TITLE
Gx updates and fixes to use new cid.Builder interface.

### PR DESCRIPTION
**Pull request by kevina**
License: MIT
Signed-off-by: Kevin Atkinson <k@kevina.org>

### CODE REVIEW

File: core/coreunix/add.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ type Adder struct {
 	mroot      *mfs.Root
 	unlocker   bstore.Unlocker
 	tempRoot   *cid.Cid
-	Prefix     *cid.Prefix
+	Prefix     cid.Builder
```
Comment (Stebalien) [inline core/coreunix/add.go]: May want to call this `CidBuilder`.

File: mfs/ops.go [hunk 2]
```diff
@@ -101,7 +101,7 @@ func PutNode(r *Root, path string, nd ipld.Node) error {
 type MkdirOpts struct {
 	Mkparents bool
 	Flush     bool
-	Prefix    *cid.Prefix
+	Prefix    cid.Builder
```
Comment (Stebalien) [inline mfs/ops.go]: Same. Should probably rename this to `CidBuilder`.

File: core/coreunix/add.go (hunk 1 above)
Comment (kevina) [reply core/coreunix/add.go]: Okay.  Fixed. I was doing this quick and planned to clean it up later.

File: mfs/ops.go (hunk 2 above)
Comment (kevina) [reply mfs/ops.go]: Ditto.

**Comment by kevina** [comment]
Failed circleci test:

```
not ok 5 - ipfs pin ls --help succeeds when input remains open
#       
#         yes | go-timeout 1 ipfs pin ls --help
#       
...
# failed 1 among 9 test(s)
1..9
make: *** [test/sharness/t0500-issues-and-regressions-offline.sh] Error 1
```

Works for me locally.

**Comment by kevina** [comment]
This should hopefully be good to go now...

**Comment by Stebalien** [comment]
@schomatis has already signed off on the unixfs PR so I'll count that as the second sign off here (no additional design decisions).

**Pull request merged**